File: app/src/main/java/com/z/palettedemo/view/ZImageView.java
package com.z.palettedemo.view;

import android.content.Context;
import android.util.AttributeSet;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatImageView;

public class ZImageView extends AppCompatImageView {
    public ZImageView(Context context) {
        super(context);
    }

    public ZImageView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public ZImageView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec);
    }
}

File: app/src/main/java/com/z/palette/view/ZImageView.kt
package com.z.palette.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

/**
 * @author by zhangkangbin
 * on 2020/5/12
 * 界面说明
 */
class ZImageView : AppCompatImageView {
    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {}

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, widthMeasureSpec)
    }
}
